SignalServiceKit/src/Messages/Interactions/OWSContactShare.m

```diff
@@ -485,55 +490,67 @@ - (void)saveAvatarImage:(UIImage *)image transaction:(YapDatabaseReadWriteTransa
     self.avatarAttachmentId = attachmentStream.uniqueId;
 }
 
-#pragma mark - Phone Numbers and Recipient IDs
+- (void)setAvatarAttachmentId:(nullable NSString *)avatarAttachmentId
+{
+    _avatarAttachmentId = avatarAttachmentId;
+}
 
-- (NSArray<NSString *> *)systemContactsWithSignalAccountPhoneNumbers:(id<ContactsManagerProtocol>)contactsManager
+@end
+
+#pragma mark -
+
+@implementation OWSContactShareProposed
```

Comment:
Instead of a subclass of our storage model, why not put the functionality here into the view model?

Reply:
There's a few reasons.  

* We want our conversion logic to deal in models, not view models.
* We want to avoid database and file system writes until necessary.

There's a bunch of related and subjective questions here, no doubt.  For one thing, I wanted to avoid a situation where our model's to be used in two completely different ways before/after being sent.